Main_Solution/main_solution_test.rb

```diff
@@ -94,24 +95,66 @@ def test_create_version
   end
 
   def test_create_issue_bug
-
+    register_user
+    create_project
     issue_options = create_issue('bug')
 
-    assert_equal(issue_options[:issue_url_slug], issue_options[:created_issue_url_slug])
+    assert_equal(issue_options[:visible_issue_id], issue_options[:created_issue_id])
   end
 
   def test_create_issue_feature
-
+    register_user
+    create_project
     issue_options = create_issue('feature')
 
-    assert_equal(issue_options[:issue_url_slug], issue_options[:created_issue_url_slug])
+    assert_equal(issue_options[:visible_issue_id], issue_options[:created_issue_id])
   end
 
   def test_create_issue_support
-
+    register_user
+    create_project
     issue_options = create_issue('support')
 
-    assert_equal(issue_options[:issue_url_slug], issue_options[:created_issue_url_slug])
+    assert_equal(issue_options[:visible_issue_id], issue_options[:created_issue_id])
+  end
+
+  def test_conditional_create_issue
+    user = register_user
+    project_name = create_project
+    random_boolean = [true, false].sample
+
+    if random_boolean
+      create_issue('bug')
+    end
+
+    navigate_to "http://demo.redmine.org/projects/#{project_name}/issues"
+
+    issue_types_list = find_elements_by_class("tracker")
+    issue_ids_list = find_elements_by_css("td.id > *:first-child")
+    issue_id_found = String.new
+
+    issue_types_list.each_with_index do |elem, index|
+      if elem.text == "Bug"
+        issue_id_found = issue_ids_list[index].text
+        navigate_to "http://demo.redmine.org/issues/#{issue_id_found}"
+        watch_icon = find_element_by_css("a.issue-#{issue_id_found}-watcher")
+        @wait.until{watch_icon.displayed?}
+        watch_icon.click
+        break
```

Comment:
@d-konovalov wanted the `each_with_index` iterator to stop because I'm looking only for the first occurrence of a "Bug" here